File: training/csharp/src/De.Sots.CellarsAndCentaurs/Domain/Service/CreatureService.cs
using De.Sots.CellarsAndCentaurs.Domain.Model;
using Microsoft.Extensions.Logging;

namespace De.Sots.CellarsAndCentaurs.Domain.Service;

public class CreatureService
{
    private readonly Creatures creatures;
    private readonly ILogger<CreatureService> logger;

    public CreatureService(Creatures creatures, ILogger<CreatureService> logger)
    {
        this.creatures = creatures;
        this.logger = logger;
    }

    public void Save(Creature creature)
    {
        logger.LogInformation("Saving creature {Id}", creature.Id);
        creatures.Save(creature);
    }

    public Creature Find(CreatureId id) => creatures.Find(id);
}

File: sample-projects/swift/Sources/CellarsAndCentaurs/Domain/Service/CreatureService.swift
final class CreatureService {
    private let creatures: Creatures

    init(creatures: Creatures) {
        self.creatures = creatures
    }

    func save(_ creature: Creature) throws {
        try creatures.save(creature)
    }

    func find(_ id: CreatureId) throws -> Creature {
        try creatures.find(id)
    }
}
